### PR TITLE
Add option to exclude base image components

### PR DESCRIPTION
**Pull request by jasonpaulos**
This PR makes two changes:

### Improve the way components from container images are mapped to layers

Prior to this, the Syft output's "Locations" field was the sole source of a component's file and layer information. This was undesirable because, for packages installed by the system package manager, the only location would be the system package manager database. As a result, any changes to system packages in later layers would result in every system package being mapped to the last layer where the database was modified.

Now, the Syft output's "Metadata.Files" field is used to augment the "Locations" data about files and layers associated with a component, and any known system package manager database files are ignored in favor of other file mappings, if present. This leads to more accurate mappings of components to image layers.

The expected result of this change is that some components which were previously mapped to layers above the base image may now be properly mapped to base image layers.

### Add a new option to exclude components which solely originate from the base image when scanning a container image.

The flag `--ExcludeBaseImageComponents` can be specified to filter out components from the base image.

### CODE REVIEW

**Review by JamieMagee — APPROVED** [review]

**Review by jpinz — APPROVED** [review]

**Review by jpinz — APPROVED** [review]

File: src/Microsoft.ComponentDetection.Orchestrator/Services/GraphTranslation/DefaultGraphTranslationService.cs [hunk 1]
```diff
@@ -41,15 +41,175 @@ public ScanResult GenerateScanResultFromProcessingResult(
...
+            return components;
+        }
+
+        // Build an indexed lookup: containerDetailId → (layerIndex → DockerLayer)
+        var layerLookup = new Dictionary<int, Dictionary<int, DockerLayer>>();
+        foreach (var (id, details) in containerDetailsMap)
+        {
+            if (details.Layers != null)
+            {
+                layerLookup[id] = details.Layers.ToDictionary(l => l.LayerIndex);
```
Comment (grvillic) [inline src/Microsoft.ComponentDetection.Orchestrator/Services/GraphTranslation/DefaultGraphTranslationService.cs]: Are we confident these layers are always unique? I have seen many times `ToDictionary` throwing errors when there happen to be two entries with the same value when we don't have much control on the source that emits this data, can we just group by and do a first/last to safely convert to dictionary?

**Review by grvillic — APPROVED** [review]

File: src/Microsoft.ComponentDetection.Orchestrator/Services/GraphTranslation/DefaultGraphTranslationService.cs (hunk 1 above)
Comment (jasonpaulos) [reply src/Microsoft.ComponentDetection.Orchestrator/Services/GraphTranslation/DefaultGraphTranslationService.cs]: Great point. I believe layers that have the same exact content will have the same ID here. I'll add proper behavior for this case

Comment (jasonpaulos) [reply src/Microsoft.ComponentDetection.Orchestrator/Services/GraphTranslation/DefaultGraphTranslationService.cs]: Actually this line is keyed on layer index, will should always be unique. Still, I added grouping logic just in case.

I also added a unit test for duplicate layer IDs to add coverage to other layer handling code.

**Review by grvillic — APPROVED** [review]

File: src/Microsoft.ComponentDetection.Detectors/linux/LinuxScanner.cs [hunk 2]
```diff
@@ -179,11 +194,16 @@ private IEnumerable<LayerMappedLinuxComponents> ProcessSyftOutputWithTelemetry(
             }
         }
 
+        // Build a file path → layerID map from the top-level files listing.
+        // This allows us to determine layer attribution for files owned by a package
+        // even when the artifact's locations only reference the package manager database.
+        var filePathToLayerId = BuildFilePathToLayerMap(syftOutput.Files);
+
```
[the remark this answers is not in the document]
Comment (jasonpaulos) [reply src/Microsoft.ComponentDetection.Detectors/linux/LinuxScanner.cs:201]: The suggestion does not provide a meaningful performance improvement, since it only optimizes for the case where the Syft output has no artifacts.
